wormhole/file_transport.go

```diff
@@ -184,40 +184,73 @@ type fileTransport struct {
 	appID           string
 }
 
-func (t *fileTransport) connectViaRelay(otherTransit *transitMsg) (net.Conn, error) {
+// combines hints and remove duplicates to minimaze connections
+func appendAndRemoveDuplicateHints(myTransit []transitHintsV1, peerTransit []transitHintsV1, hintType string) []transitHintsRelay {
```

Comment:
Using a set datastructure instead of lists would make this entire function redundant, and perform faster.

Also, why are we filtering for `hintType` here?

Reply:
Because we have two different functions: connectDirect and connectViaRelay, which needs only specific hints to iterate. So either we filter in specific function either we pass parameter to filter on function.

Reply:
Also, not all hints should be merged, like hints for direct tcp connection (not via tcp relay).

Reply:
Okay. How about `mergeFilterHints` as function name then?

Reply:
Yes, set is the right data structure here. Go doesn't have an inbuilt set type, however, we can implement set using `map[T]struct{}` or something of that sort. The `T` in our case would be `transitHintsRelay`. We can use any type for `T` as long as `==` or `!=` is defined for that type. So, if a `struct` has a slice for instance, we cannot use that `struct` as `T`.

Reply:
changed